src/test/java/eu/qwan/vender/VendingMachineTest.java

```diff
@@ -22,19 +22,35 @@ public void deliversNothing() {
         }
     }
 
-    @Test
-    public void deliversCanOfChoice() {
-        assertEquals(Optional.of(Can.COLA), machine.deliver(Choice.COLA));
-        assertEquals(Optional.of(Can.FANTA), machine.deliver(Choice.FANTA));
-        assertEquals(Optional.of(Can.SPRITE), machine.deliver(Choice.SPRITE));
-    }
+    @Nested
+    class FreeMachine {
```

Comment:
Nice! I didn't know @Nested yet so I learned something new today :) 
Naming suggestion: you could name the nested test class for example: "WhenFree" so that it looks more like a behaviour specification of the vending machine.